src/XMLSecurityKey.php

```diff
@@ -375,7 +368,7 @@ private function encryptMcrypt($data)
         if ($this->cryptParams['mode'] == MCRYPT_MODE_CBC) {
             $bs = mcrypt_enc_get_block_size($td);
             for ($datalen0 = $datalen = strlen($data); (($datalen % $bs) != ($bs - 1)); $datalen++)
```

Comment:
Looping over `random_int` for character codes is a little overkill. Not 100% sure what's going on here, looks like adding randomised padding? Do you know the reason for the selected character set? (Why hasn't a standard padding mode been used?)